flytepropeller/pkg/controller/executors/kube.go

```diff
@@ -33,107 +33,91 @@ var NewCache = func(config *rest.Config, options cache.Options) (cache.Cache, er
 	return otelutils.WrapK8sCache(k8sCache), nil
 }
 
-var NewClient = func(config *rest.Config, options client.Options) (client.Client, error) {
-	var reader *fallbackClientReader
-	if options.Cache != nil && options.Cache.Reader != nil {
-		// if caching is enabled we create a fallback reader so we can attempt the client if the cache
-		// reader does not have the object
-		reader = &fallbackClientReader{
-			orderedClients: []client.Reader{options.Cache.Reader},
+func BuildNewClientFunc(writeFilterSize int, scope promutils.Scope) func(config *rest.Config, options client.Options) (client.Client, error) {
+	return func(config *rest.Config, options client.Options) (client.Client, error) {
+		var cacheReader client.Reader
+		cachelessOptions := options
+		if options.Cache != nil && options.Cache.Reader != nil {
+			cacheReader = options.Cache.Reader
+			cachelessOptions.Cache = nil
 		}
 
-		options.Cache.Reader = reader
-	}
+		k8sClient, err := client.New(config, cachelessOptions)
+		if err != nil {
+			return k8sClient, err
```

Comment:
It doesn't seem to be reasonable to return `k8sClient`, which could potentially be something built partially if `client.New` does something weird.

```suggestion
			return nil, err
```

Reply:
done.